### DISCUSSION TITLE
ApostropheCMS database migration

### DISCUSSION THREAD

**Discussion by sumithra16** [apostrophecms/apostrophe#2714] [Q&A] [answered]
I need some help on apostrophe content migration!
I am trying to move my content from my local to staging with apos.migrate method in that AddTask method to clone apostrophe content. But I couldn't see any error or changes reflected to staging site.
node app apostrophe-migrations:migrate
Is there a way for database migration from DB to DB? Any small help would be much appreciated.

**Comment by myovchev** [comment, marked as answer]
I don't understand the question. Migrations API is used to migrate changes of your database schema (on one and the same server) and not moving data from one server to another. So the flow will be you do local improvements (developments), you reflect your DB changes as migration scripts, you deploy your new code to the staging server (with your favorite deploy method) and in the end you execute `node app apostrophe-migrations:migrate` on your staging server.

If I understand your problem correctly, you need initial dump from your local DB and initial restore on your staging DB which are described  in details in [the MongoDB documentation](https://docs.mongodb.com/manual/tutorial/backup-and-restore-tools/)

**Reply by sumithra16** [reply]
Yup,Useful. Thank you! @myovchev

**Comment by boutell** [comment]
Yes, if your goal is to move *content* between *servers*, you should use
"mongodump" and "mongorestore". You also need to rsync the public/uploads
folder. And there are sample scripts for it in the "scripts" folder
(assuming A2 and a project born from apostrophe-boilerplate).

On Mon, Feb 8, 2021 at 11:35 AM Miro Yovchev <notifications@github.com>
wrote:

> I don't understand the question. Migrations API is used to migrate changes
> of your database schema (on one and the same server) and not moving data
> from one server to another. So the flow will be you do local improvements
> (developments), you reflect your DB changes as migration scripts, you
> deploy your new code to the staging server (with your favorite deploy
> method) and in the end you execute node app apostrophe-migrations:migrate
> on your staging server.
>
> If I understand your problem correctly, you need initial dump from your
> local DB and initial restore on your staging DB which are described in
> details in the MongoDB documentation
> <https://docs.mongodb.com/manual/tutorial/backup-and-restore-tools/>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/2714#discussioncomment-350006>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27O2HBB2OASGWDDGL7TS6AHEJANCNFSM4XIZIJGQ>
> .
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Reply by sumithra16** [reply]
Thank you! @boutell
